Realm/RLMArrayLinkView.mm

```diff
@@ -427,3 +441,48 @@ - (RLMNotificationToken *)addNotificationBlock:(void (^)(RLMArray *, RLMCollecti
 #pragma clang diagnostic pop
 
 @end
+
+@interface RLMArrayLinkViewHandoverMetadata : NSObject
+
+@property (nonatomic) NSString *parentClassName;
+@property (nonatomic) NSString *key;
+
+@end
+
+@implementation RLMArrayLinkViewHandoverMetadata
+@end
+
+@interface RLMArrayLinkView (ThreadConfined) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMArrayLinkView (ThreadConfined)
```

Comment:
I'm not sure why this is a category. Splitting chunks of the implementation off into separate categories isn't something we do anywhere else in obj-c and it doesn't have obvious organizational benefits.